libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/maneuver/ManeuverViewHelper.kt

```diff
@@ -0,0 +1,178 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation.maneuver
+
+import androidx.core.util.Pair
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_LEFT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_RIGHT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_SHARP_LEFT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_SHARP_RIGHT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_SLIGHT_LEFT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_SLIGHT_RIGHT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_STRAIGHT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_UTURN
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_ARRIVE
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_EXIT_ROTARY
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_EXIT_ROUNDABOUT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_FORK
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_MERGE
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_OFF_RAMP
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_ROTARY
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_ROUNDABOUT
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_ROUNDABOUT_TURN
+
+object ManeuverViewHelper {
+
+    const val DEFAULT_ROUNDABOUT_ANGLE = 180f
+
+    private const val TOP_ROUNDABOUT_ANGLE_LIMIT = 300f
+    private const val BOTTOM_ROUNDABOUT_ANGLE_LIMIT = 60f
+
+    @JvmField
+    val MANEUVER_VIEW_UPDATE_MAP: Map<Pair<String, String>, ManeuverViewUpdate> = object : HashMap<Pair<String, String>, ManeuverViewUpdate>() {
```

Comment:
may be replace `Pair` by custom `model`?

Reply:
Using a custom `model` would be easier readable here. But I don't see too much benefit. Since it's the current implementation and it works well, I prefer to keep it for now.